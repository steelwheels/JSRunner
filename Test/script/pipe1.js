/**
 * pipe1.js
 */

let pipe = Pipe() ;

console.log("[allocate process]\n") ;
let process = system("/bin/echo \"Hello, world !!\"",
					stdin, pipe.writing, stderr) ;

if(process == null){
	console.log("[Error] Could not launch command\n") ;
	exit(1) ;
}

console.log("Wait until exit\n") ;
process.waitUntilExit() ;

//console.log("Close write pipe\n") ;
//pipe.writing.close() ;

/* receive output */
let reader = pipe.reading ;
//console.log("[receive output] " + reader + "\n") ;
let c = reader.getc() ;
while(c != null){
	console.log(`[receive] ${c}\n`) ;
	c = reader.getc() ;
}

console.log("[bye]\n");

