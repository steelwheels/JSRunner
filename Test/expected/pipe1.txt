[allocate process]
Wait until exit
[receive] H
[receive] e
[receive] l
[receive] l
[receive] o
[receive] ,
[receive]  
[receive] w
[receive] o
[receive] r
[receive] l
[receive] d
[receive]  
[receive] !
[receive] !
[receive] 

[bye]
